.github/workflows/storybook.yml

```diff
@@ -1,25 +1,34 @@
-name: Deploy Storybook
+name: Build and Deploy Storybook
+
 on:
+  pull_request:
+    branches:
+      - '*'
   push:
     branches:
       - master
+
 jobs:
-  build-and-deploy:
+  build_and_deploy:
```

Comment:
can we split in 2 jobs ?
when opening a PR we only want to build, to make sure it's not broken

we only want to build and deploy when merging a PR to main